### PR TITLE
Smörgåsbord

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/code_a5dd0_len_114e0.c [hunk 1]
```diff
@@ -116,7 +116,19 @@ INCLUDE_ASM(s32, "code_a5dd0_len_114e0", AssignAreaFlag, ScriptInstance* script,
 
 INCLUDE_ASM(s32, "code_a5dd0_len_114e0", AssignBlockFlag, ScriptInstance* script, s32 isInitialCall);
 
-INCLUDE_ASM(s32, "code_a5dd0_len_114e0", AssignFlag, ScriptInstance* script, s32 isInitialCall);
+//INCLUDE_ASM(s32, "code_a5dd0_len_114e0", AssignFlag, ScriptInstance* script, s32 isInitialCall);
```
Comment (pixel-stuck) [inline src/code_a5dd0_len_114e0.c:119]: probably should be removed

File: src/code_e92d0_len_5da0.c [hunk 2]
```diff
@@ -673,13 +858,65 @@ ApiStatus si_handle_call(ScriptInstance* script) {
...
+    ScriptInstance* newScript;
+    s32 i;
+
+    newScript = start_script_in_group((ScriptInstance*)get_variable(script, *script->ptrReadPos), script->priority, 0,
+                                      script->groupFlags);
+
+    newScript->ownerActorID = script->ownerActorID;
+    newScript->ownerID = script->ownerID;
+
+    i = 0;
```
Comment (pixel-stuck) [inline src/code_e92d0_len_5da0.c:871]: the loops here are very weird... I assume they don't match as for loops? Seems kinda weird.

File: src/code_e92d0_len_5da0.c [hunk 3]
```diff
@@ -754,32 +1046,60 @@ ApiStatus si_handle_resume(ScriptInstance* script) {
...
-        script->labelIndices[1] = NULL;
-        script->flags.flags &= ~0x2;
+    if (!does_script_exist(trigger->runningScriptID)) {
+        trigger->runningScript = NULL;
+        trigger->flags.flags &= ~0x2;
     }
 }
 
 INCLUDE_ASM(s32, "code_e92d0_len_5da0", si_handle_bind_lock, ScriptInstance* script, s32 isInitialCall);
+/*ApiStatus si_handle_bind_lock(ScriptInstance* script) {
```
Comment (pixel-stuck) [inline src/code_e92d0_len_5da0.c:1075]: what's going on with this?

**Review by pixel-stuck — CHANGES_REQUESTED** [review]

File: src/code_1f580_len_1940.c [hunk 4]
```diff
@@ -77,70 +105,352 @@ ApiStatus SetBattleMusic(ScriptInstance* script, s32 isInitialCall) {
...
+    Bytecode* args = script->ptrReadPos;
+    Enemy* owner = script->ownerActorID;
+    NpcId npcId = get_variable(script, *args++);
+    Bytecode* auxBytecode = (Bytecode*)get_variable(script, *args);
+
+    if (npcId == NpcId_SELF) {
+        npcId = owner->npcID;
+    }
+
+    get_enemy(npcId)->auxBytecode = auxBytecode;
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_1f580_len_1940.c:191]: I resolved this but then took another look and am not sure I agree. I think it's much more readable to omit storing the result into that temp whenever possible.

File: src/code_1f580_len_1940.c [hunk 5]
```diff
@@ -77,70 +105,352 @@ ApiStatus SetBattleMusic(ScriptInstance* script, s32 isInitialCall) {
...
     script->ownerActorID->flags = *script->ptrReadPos;
     return ApiStatus_DONE2;
 }
 
 ApiStatus SetSelfEnemyFlagBits(ScriptInstance* script, s32 isInitialCall) {
-    Bytecode* thisPos = script->ptrReadPos;
+    Bytecode* args = script->ptrReadPos;
     Enemy* owner = script->ownerActorID;
-    s32 bits = thisPos[0];
+    s32 bits = args[0];
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_1f580_len_1940.c:381]: Yeah, the codegen is different but that does indeed work!

File: src/code_1f580_len_1940.c [hunk 6]
```diff
@@ -164,7 +474,23 @@ ApiStatus func_8004580C(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "code_1f580_len_1940", func_80045838);
+//INCLUDE_ASM(s32, "code_1f580_len_1940", func_80045838);
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_1f580_len_1940.c:477]: No, just forgart

File: src/code_F5750.c [hunk 7]
```diff
@@ -48,28 +48,39 @@ ApiStatus DisablePlayerInput(ScriptInstance* script, s32 isInitialCall) {
...
-    Npc* playerNpc = gPlayerNpc;
 
-    playerNpc->pos.x = x;
-    playerNpc->pos.z = z;
-    playerNpc->pos.y = y;
+    gPlayerNpcPtr->pos.x = x;
+    gPlayerNpcPtr->pos.y = y;
+    gPlayerNpcPtr->pos.z = z;
+
+    playerStatus->position.x = gPlayerNpcPtr->pos.x;
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/code_F5750.c:62]: no, unfortunately

File: src/code_e92d0_len_5da0.c (hunk 3 above)
Comment (ethteck) [reply src/code_e92d0_len_5da0.c:1075]: Left in by accident

File: src/code_e92d0_len_5da0.c (hunk 2 above)
Comment (ethteck) [reply src/code_e92d0_len_5da0.c:871]: yeah, I thought they didn't work normally but I guess I was wrong

**Pull request merged**